src/annotator/sidebar.tsx

```diff
@@ -187,6 +189,12 @@ export class Sidebar implements Destroyable {
       shadowRoot.appendChild(this.iframeContainer);
 
       element.appendChild(this._hypothesisSidebar);
+
+      // Render a container for toast messages in the host frame. The sidebar
+      // will forward messages to render here while it is collapsed.
+      const messagesElement = document.createElement('div');
+      shadowRoot.appendChild(messagesElement);
+      render(<ToastMessages sidebarRPC={this._sidebarRPC} />, messagesElement);
```

Comment:
I think this would not be necessary, because the `messagesElement` is part of `this._hypothesisSidebar`'s shadow DOM, and we remove `this._hypothesisSidebar` entirely on `destroy`: https://github.com/hypothesis/client/blob/61273c7741bee8e4b3d68d2a0db6a97757de79b8/src/annotator/sidebar.ts#L270

Reply:
Oh, I think I get what you mean. So that the useEffect subscriptions are cleaned-up. Gotcha.